KCRO-115: add a field to view result indicating of a user doc has photo

oguerrero913 opened the pull request:
added valitation to views
 - crossbar_listing
 - list_by_id
to check whether a user's doccument has or not a photo attached

oguerrero913 committed:
added valitation to view
 - crossbar_listing
 - list_by_id
to check whether a user's doccument has or not a photo attached

icehess reviewed (changes requested):
`users.json` is the old view, we need to update `account-crossbar_listings.json` view too, this the view we use in `cb_users`.

When changing `account-crossbar_listings.json`, add this logic to all places that is creating `features` field (in `by_type_id`, `by_type_name` and `by_type_ownerid`).

Other than that, LGTM.

oguerrero913 committed:
added photo validation to account-crossbar_listing.json in views:
- by_type_id
- by_type_name
- by_type_ownerid

oguerrero913 committed:
added functions avatar/1, set_avatar/2

oguerrero913 committed:
changed function's names:
- avatar/1 -> has_avatar/1
- avatar/2 -> has_avatar/2
- set_avatar -> set_has_avatar/2
fixed spec for has_avatar/1 and has_avatar/2

oguerrero913 committed:
fixed spec

icehess committed:
Merge branch 'master' into master-KCRO115-Add-avatar-validation
